### PR TITLE
updating to elastic 7.x and activate analyzers for default properties

### PR DESCRIPTION
**Pull request by flavico**
Tests were fix because when the the search in catalog by properties the return come by json format. The places where filters didn't matter, I removed.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import os
 
 
-version = '3.0.5.dev0'
+version = '3.0.5'
```
Comment (idgserpro) [inline setup.py]: @flavico please don't remove `dev0`

File: docs/history.rst [hunk 2]
```diff
@@ -4,8 +4,8 @@ Changelog
 3.0.5 (unreleased)
 ------------------
 
-- Fix commit hook bug when content has been moved
-  [instification]
```
Comment (idgserpro) [inline docs/history.rst]: @flavico please don't remove the previous entry. Add yours above the previous one.

File: docs/history.rst [hunk 3]
```diff
@@ -4,8 +4,8 @@ Changelog
...
+- Update to work in Elastic Search version 7.x (testing in 7.4 and 7.5 works fine)
+- Fix to create index in ES with analyzed properties to the default plone fields like (SearchableText, title, etc...)
```
Comment (idgserpro) [inline docs/history.rst]: Add your id:

`[flavico]`

File: collective/elasticsearch/tests/test_search.py [hunk 4]
```diff
@@ -121,43 +122,42 @@ def test_path_index_query(self):
 
         self.commit()
         self.es.connection.indices.flush()
+        time.sleep(1)
 
         self.assertEqual(
-            len(self.catalog(path={'depth': 0, 'query': '/plone/folder0'},
```
Comment (idgserpro) [inline collective/elasticsearch/tests/test_search.py]: `len` should work on catalog result.

File: collective/elasticsearch/tests/test_search.py [hunk 5]
```diff
@@ -21,27 +21,27 @@ def test_field_index_query(self):
         createObject(self.portal, 'Event', 'event', title='Some Event')
         self.commit()
         self.es.connection.indices.flush()
-        el_results = self.catalog(portal_type='Event', Title='some event')
```
Comment (idgserpro) [inline collective/elasticsearch/tests/test_search.py]: Why did you remove the title?

Comment (flavico) [reply collective/elasticsearch/tests/test_search.py]: Cause, when you search by "Title", the new api from ES7, bring the amount on a json format. Intead of when you search without "Title", the amount comes in "integer".
In this case, the test by the "Title" is not necessary, because only one event has created and search by type.

File: collective/elasticsearch/tests/test_search.py (hunk 4 above)
Comment (flavico) [reply collective/elasticsearch/tests/test_search.py]: In this case, works on "len" cause the result is "integer"

File: docs/history.rst (hunk 3 above)
Comment (idgserpro) [reply docs/history.rst]: You haven't done that yet.

File: docs/history.rst (hunk 2 above)
Comment (idgserpro) [reply docs/history.rst]: You haven't done that yet

File: collective/elasticsearch/mapping.py [hunk 6]
```diff
@@ -63,7 +63,9 @@ def __call__(self):
         for name in self.catalog.indexes.keys():
             index = getIndex(self.catalog, name)
             if index is not None:
-                properties[name] = index.create_mapping(name)
+                # prevent create default mapping analyzers
+                if name not in properties:
```
Comment (idgserpro) [inline collective/elasticsearch/mapping.py]: Why is this necessary? Is it necessary for ES7 or does ES6 have this problem too?

File: collective/elasticsearch/tests/test_search.py (hunk 5 above)
Comment (idgserpro) [reply collective/elasticsearch/tests/test_search.py]: Whatever the return of ES, it should be possible to search by Title. Try change the implementation, rather than changing the test.

File: setup.py (hunk 1 above)
Comment (idgserpro) [reply setup.py]: You haven't done that yet.

File: collective/elasticsearch/tests/test_search.py (hunk 4 above)
Comment (idgserpro) [reply collective/elasticsearch/tests/test_search.py]: It must be possible to use `len`, regardless of how the ES returns the result.

File: collective/elasticsearch/mapping.py (hunk 6 above)
Comment (flavico) [reply collective/elasticsearch/mapping.py]: It's a problem with occur in ES6 and ES7. 
To create mapping with the fields analyzed and tokenized. The code currently ignores the mapping configuring and apply without configuration.

File: collective/elasticsearch/tests/test_search.py (hunk 5 above)
Comment (flavico) [reply collective/elasticsearch/tests/test_search.py]: It's possible to do this, but to simplified the test, i removed the filter by Title cause there's only one object. In other cases with many records i used this code "._len['value']".

File: collective/elasticsearch/tests/test_search.py (hunk 4 above)
Comment (flavico) [reply collective/elasticsearch/tests/test_search.py]: Yes, if you filter the result, the new version of ES bring the amount e how the amount is exact.

File: collective/elasticsearch/mapping.py (hunk 6 above)
Comment (idgserpro) [reply collective/elasticsearch/mapping.py]: ok

File: collective/elasticsearch/tests/test_search.py (hunk 5 above)
Comment (idgserpro) [reply collective/elasticsearch/tests/test_search.py]: Please change the implementation not the tests.

File: collective/elasticsearch/tests/test_search.py (hunk 4 above)
Comment (idgserpro) [reply collective/elasticsearch/tests/test_search.py]: Please change the implementation not the tests

File: collective/elasticsearch/mapping.py (hunk 6 above)
Comment (idgserpro) [reply collective/elasticsearch/mapping.py]: I took a look and I still don't understand the need for this test. The current code overrides the default configuration by index type configuration. Why use the default configuration and not the index type configuration? Can you give an example of a problem index?

**Comment by idgserpro** [comment]
Replaced by #78
